rust/stracciatella_c_api/src/c/fs.rs

```diff
@@ -0,0 +1,209 @@
+//! This module contains the C interface for [`stracciatella::fs`].
+//!
+//! [`stracciatella::fs`]: ../../stracciatella/fs/index.html
+
+use std::f64;
+use std::ptr;
+use std::time;
+
+use stracciatella::fs;
+
+use crate::c::common::*;
+use crate::c::misc::VecCString;
+
+/// A directory in the filesystem that is automatically deleted.
+/// The contents of the directory are deleted before the directory is deleted.
+pub struct TempDir {
```

Comment:
Should we extract TempDir into its own file and make it independent from `Fs`?  As `Fs_createTempDir` is also not really dependent on `Fs`.

On second thought, is it really necessary do define our own `TempDir`? Can't we just return the one from the crate and define the `TempDir_` functions on that one?

Reply:
It relies on the filesystem functions to create a special type of directory, so I think it qualifies for `Fs`.

Unfortunately `cbindgen` won't emit an opaque type for the original `TempDir`, so the wrapper is needed.